### PR TITLE
Collab serial ble ot

### PR DESCRIPTION
**Pull request by canisLupus1313**

### CODE REVIEW

File: applications/protocols_serialization/client/snippets/debug/debug.conf [hunk 1]
```diff
@@ -20,7 +20,8 @@ CONFIG_LOG_MODE_DEFERRED=y
 CONFIG_LOG_MAX_LEVEL=4
 
 CONFIG_LOG_BACKEND_RTT=n
-CONFIG_SHELL_LOG_BACKEND=y
+#CONFIG_SHELL_LOG_BACKEND=y
```
Comment (Damian-Nordic) [inline applications/protocols_serialization/client/snippets/debug/debug.conf:23]: Why this change? I guess if both logger and shell write to UART, we may expect some interwoven lines and that's why you typically use shell logging backend when shell is enabled.

File: subsys/nrf_rpc/nrf_rpc_uart.c [hunk 2]
```diff
@@ -66,47 +66,62 @@ static void work_handler(struct k_work *work)
...
+		rx_len = ring_buf_put_claim(
+			&uart_tr->rx_ringbuf, &rx_buffer,
+			uart_tr->rx_ringbuf.size);
+		if (rx_len > 0) {
+			rx_len = uart_fifo_read(uart, rx_buffer, rx_len);
+			int err = ring_buf_put_finish(
+				&uart_tr->rx_ringbuf, rx_len);
+			if (err != 0) {
+				__ASSERT_NO_MSG(err == 0);
+			}
```
Comment (Damian-Nordic) [inline subsys/nrf_rpc/nrf_rpc_uart.c:107]:
```suggestion
			__ASSERT_NO_MSG(err == 0);
```

File: subsys/nrf_rpc/nrf_rpc_uart.c [hunk 3]
```diff
@@ -66,47 +66,62 @@ static void work_handler(struct k_work *work)
...
-				rx_len, ret);
-			break;
+			/* No space in the ring buffer - consume byte. */
+			LOG_WRN("RX ring buffer full.");
+
+			rx_len = uart_fifo_read(uart, &dummy, 1);
 		}
 
-		k_work_submit_to_queue(&uart_tr->rx_workq, &uart_tr->cb_work);
+		if (new_data) {
```
Comment (Damian-Nordic) [inline subsys/nrf_rpc/nrf_rpc_uart.c:122]: nit: You might call this after the loop so that `k_work` is always scheduled only once per ISR.

**Review by Damian-Nordic — APPROVED** [review]
Looks great.

File: applications/protocols_serialization/client/snippets/debug/debug.conf (hunk 1 above)
Comment (canisLupus1313) [reply applications/protocols_serialization/client/snippets/debug/debug.conf:23]: I can change it to a comment. But as it turns out when we are processing the shell command the logs aren't printed out since shell is busy. moving it to UART backend breaks formatting a bit but do not delay logging.

File: subsys/nrf_rpc/nrf_rpc_uart.c (hunk 3 above)
Comment (canisLupus1313) [reply subsys/nrf_rpc/nrf_rpc_uart.c]: fixed.

**Review by kszewenko — APPROVED** [review]

**Review by Damian-Nordic — APPROVED** [review]
